jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/core/entityManager2/Client3.java

```diff
@@ -0,0 +1,361 @@
+        String pkgName = pkgNameWithoutSuffix + ".";
+        String[] classes = {pkgName + "DoesNotExist", pkgName + "Employee", pkgName + "Order"};
+        return createDeploymentJar("jpa_core_entityManager3.jar", pkgNameWithoutSuffix, classes);
+
+    }
+
+    /*
+     * setupOrderData() is called before each test
+     *
+     * @class.setup_props: jdbc.db;
+     */
+    @BeforeEach
+    public void setupOrderData() throws Exception {
+        logger.log(Logger.Level.TRACE, "setupOrderData");
+        try {
+            super.setup();
+            createDeployment();
+            removeTestData();
+            createOrderData();
+            map.putAll(getEntityManager().getProperties());
+            map.put("foo", "bar");
+            displayMap(map);
+            dataBaseName = System.getProperty("jdbc.db");
+        } catch (Exception e) {
+            logger.log(Logger.Level.ERROR, "Exception: ", e);
+            throw new Exception("Setup failed:", e);
+        }
+    }
+
+    @AfterEach
+    public void cleanupData() throws Exception {
+        try {
+            logger.log(Logger.Level.TRACE, "Cleanup data");
+            removeTestData();
+            cleanup();
+        } finally {
+            removeTestJarFromCP();
+        }
+    }
+
+    @Test
+    public void getReferenceForExistingEntityTest() throws Exception {
+        boolean pass1 = false;
+        boolean pass2 = false;
+        try {
+            getEntityTransaction().begin();
+            Order order = new Order(1, 111, "desc1");
+            Order reference = getEntityManager().getReference(order);
+            // Verify that access to entity attribute works
+            String orderDescription = reference.getdescription();
+            getEntityTransaction().commit();
+
+            if (reference instanceof Order) {
+                pass1 = true;
+            } else {
+                logger.log(Logger.Level.ERROR, "Fetched entity is not same as expected.");
+            }
+            if (orderDescription != null) {
+                pass2 = true;
+            } else {
+                logger.log(Logger.Level.ERROR, "Access to entity attribute doesn't work.");
+            }
+
+        } catch (Exception e) {
+            getEntityTransaction().rollback();
+            logger.log(Logger.Level.ERROR, "Unexpected exception occurred", e);
+        }
+        if (!pass1 || !pass2) {
+            throw new Exception("getReferenceForExistingEntityTest failed");
+        }
+    }
+
+    @Test
+    public void getReferenceForNonExistingEntityTest() throws Exception {
+        boolean pass = false;
+        try {
+            Order order = new Order(0, 000, "desc0");
+            // Verify that getReference() to order fails
+            // EntityNotFoundException shall be thrown on non-existing entity access
+            Order reference = getEntityManager().getReference(order);
+        } catch (EntityNotFoundException enfe) {
```

Comment:
this exception is not required to be thrown directly from the `getReference` method, the spec allows this to be thrown later, when some property of the entity is accessed.